expression/integration_test.go

```diff
@@ -1366,6 +1366,21 @@ func (s *testIntegrationSuite) TestBuiltin(c *C) {
 	result = tk.MustQuery("select cast('-34 100:00:00' as time);")
 	result.Check(testkit.Rows("-838:59:59"))
 
+	// fix issue #4324
+	invalidTimeCastSqls := []string{
+		"select cast('239010' as time);",
+		"select cast('23:90:10' as time);",
+		"select cast('233070' as time);",
+		"select cast('23:30:70' as time);",
+	}
+	for _, invalidTimeCastSql := range invalidTimeCastSqls {
+		rs, err := tk.Exec(invalidTimeCastSql)
+		c.Assert(errors.ErrorStack(err), Equals, "")
+		c.Assert(rs, NotNil)
+		_, err = tidb.GetRows(rs)
+		c.Assert(err.Error(), Equals, "invalid time format")
```

Comment:
in mysql, the behavior of invalid time format is depended on sql mode, in strict sql mode:
```sql
MySQL > select cast('239010' as time);
+------------------------+
| cast('239010' as time) |
+------------------------+
| NULL                   |
+------------------------+
1 row in set, 1 warning (0.00 sec)

MySQL > show warnings;
+---------+------+------------------------------------------+
| Level   | Code | Message                                  |
+---------+------+------------------------------------------+
| Warning | 1292 | Truncated incorrect time value: '239010' |
+---------+------+------------------------------------------+
1 row in set (0.00 sec)
```

and:
```sql
MySQL > create table t(a time);
Query OK, 0 rows affected (0.03 sec)

MySQL > insert into t select cast('239010' as time);
ERROR 1292 (22007): Truncated incorrect time value: '239010'
```

Reply:
Please use `session.go:HandleTruncate` to handle it.

Reply:
ok. done. PTAL.